Sources/XcodeProj/Objects/Project/PBXObjectReference.swift

```diff
@@ -1,72 +1,102 @@
 import Foundation
 
 /// Object used as a reference to PBXObjects from PBXObjects.
-class PBXObjectReference: NSObject, Comparable, NSCopying {
+class PBXObjectReference: NSObject, Comparable, NSCopying, @unchecked Sendable {
+    private let lock = NSRecursiveLock()
+    
     /// Boolean that indicates whether the id is temporary and needs
     /// to be regenerated when saving it to disk.
-    private(set) var temporary: Bool
+    var temporary: Bool {
+        lock.withLock { _temporary }
+    }
+    private var _temporary: Bool
 
     /// String reference.
-    private(set) var value: String
+    var value: String {
+        lock.withLock { _value }
+    }
+    private var _value: String
+    
 
     /// Weak reference to the objects instance that contains the project objects.
-    weak var objects: PBXObjects?
+    var objects: PBXObjects? {
+        get {
+            lock.withLock { _objects }
+        }
+        set {
+            lock.withLock {
+                _objects = newValue
+            }
+        }
+    }
+    private weak var _objects: PBXObjects?
 
+    // QUESTION: this is exposed to the project but so is `getThrowingObject` and `getObject`.  What access patterns do we want to support?
```

Comment:
Good point. I think having three APIs for the same is unnecessary. I believe (it's been a long time) that the motivation for having `getThrowingObject` was to instruct developers at runtime about bad usages of the library. For example adding a reference to an object that's not strongly referenced by the project. However, looking at the `getThrowingObject` implementation, and don't know if the error that we are throwing is more helpful than trying to do a force unwrap of the referenced object. 
I'm between making `getThrowingObject` errors more useful, which would require adding more context to `PBXObjectReference` to ensure it's in sync, which is a bit of a tricky challenge, or removing `getObject` and `getThrowingObject` and having a single accessor. I'd say that we do the latter.